src/gpu_sysman_test.c

```diff
@@ -842,120 +861,6 @@ int plugin_dispatch_metric_family(metric_family_t const *fam) {
- * instead of increasing it one-by-one, like the real collectd metrics
- * code does
- */
-int metric_label_set(metric_t *m, char const *name, char const *value) {
-  assert(m && name);
-  size_t num = m->label.num;
-  label_pair_t *pair = m->label.ptr;
-  if (num) {
-    assert(num < MAX_LABELS);
-    assert(pair);
-  } else {
-    assert(!pair);
-    pair = scalloc(MAX_LABELS, sizeof(*pair));
-    m->label.ptr = pair;
-  }
-  int i;
-  for (i = 0; i < MAX_LABELS; i++) {
-    if (!pair[i].name) {
-      /* not found -> new label */
-      pair[i].name = sstrdup(name);
-      m->label.num++;
-      break;
-    }
-    if (strcmp(name, pair[i].name) == 0) {
-      break;
-    }
-  }
-  assert(value); /* removing label with NULL 'value' is not supported */
-  free(pair[i].value);
-  pair[i].value = sstrdup(value);
-  return 0;
-}
-
-int metric_reset(metric_t *m) {
-  assert(m);
-  size_t num = m->label.num;
-  label_pair_t *pair = m->label.ptr;
-  if (!num) {
-    assert(!pair);
-    return 0;
-  }
-  assert(pair);
-  for (int i = 0; i < MAX_LABELS; i++) {
-    if (!pair[i].name) {
-      break;
-    }
-    free(pair[i].name);
-    free(pair[i].value);
-    pair[i].value = pair[i].name = NULL;
-    num--;
-  }
-  assert(!num);
-  free(pair);
-  m->label.ptr = NULL;
-  m->label.num = 0;
-  return 0;
-}
-
-#define MAX_METRICS 8
-
-/* mock function uses just one large enough metrics array (for testing)
- * instead of increasing it one-by-one, like the real collectd metrics
- * code does
- */
-int metric_family_metric_append(metric_family_t *fam, metric_t m) {
-  assert(fam);
-  size_t num = fam->metric.num;
-  metric_t *metric = fam->metric.ptr;
-  if (num) {
-    assert(num < MAX_METRICS);
-    assert(metric);
-  } else {
-    assert(!metric);
-    metric = scalloc(MAX_METRICS, sizeof(*metric));
-    fam->metric.ptr = metric;
-  }
-  /* copy metric and pointers to its labels */
-  metric[num] = m;
-  label_pair_t *src = m.label.ptr;
-  if (src) {
-    /* alloc max size as labels can be added also to family metrics copies */
-    label_pair_t *dst = scalloc(MAX_LABELS, sizeof(*src));
-    metric[num].label.ptr = dst;
-    for (size_t i = 0; i < m.label.num; i++) {
-      dst[i].name = sstrdup(src[i].name);
-      dst[i].value = sstrdup(src[i].value);
-    }
-  }
-  fam->metric.num++;
-  m.family = fam;
-  return 0;
-}
-
-int metric_family_metric_reset(metric_family_t *fam) {
-  metric_t *metric = fam->metric.ptr;
-  for (size_t m = 0; m < fam->metric.num; m++) {
-    label_pair_t *pair = metric[m].label.ptr;
-    for (size_t i = 0; i < metric[m].label.num; i++) {
-      free(pair[i].name);
-      free(pair[i].value);
-    }
-    free(pair);
-    metric[m].label.ptr = NULL;
-    metric[m].label.num = 0;
-  }
-  free(fam->metric.ptr);
-  fam->metric.ptr = NULL;
-  fam->metric.num = 0;
-  return 0;
-}
-
 /* ------------------------------------------------------------------------- */
 /* mock up of collectd plugin API */
```

Comment:
Would it make sense to move this to `src/daemon/plugin_mock.c`?

Reply:
I need the information passed to that API, to validate correct working of the plugin's config and metric handling. What is checked from metric and logging parts in the mocked functions is specific to sysman plugin, config options less so, but I think it makes sense to keep it all plugin specific.

Reply:
Sure, works for me.